src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -68,24 +72,31 @@ namespace Opm {
             aqurecords.at(aquanconRecordIdx).k1 = aquanconRecord.getItem("K1").template get<int>(0);
             aqurecords.at(aquanconRecordIdx).k2 = aquanconRecord.getItem("K2").template get<int>(0);
 
-            aquiferID_per_record.at(aquanconRecordIdx) = aquanconRecord.getItem("AQUIFER_ID").template get<int>(0);
             m_maxAquID = (m_maxAquID < aquiferID_per_record.at(aquanconRecordIdx) )?
                             aquiferID_per_record.at(aquanconRecordIdx) : m_maxAquID;
 
             double influx_mult = aquanconRecord.getItem("INFLUX_MULT").getSIDouble(0);
 
             FaceDir::DirEnum faceDir = FaceDir::FromString(aquanconRecord.getItem("FACE").getTrimmedString(0));
 
+            const auto& properties = eclstate.get3DProperties();
             // Loop over the cartesian indices to convert to the global grid index
             for (int k=aqurecords.at(aquanconRecordIdx).k1; k <= aqurecords.at(aquanconRecordIdx).k2; k++) {
                 for (int j=aqurecords.at(aquanconRecordIdx).j1; j <= aqurecords.at(aquanconRecordIdx).j2; j++)
-                    for (int i=aqurecords.at(aquanconRecordIdx).i1; i <= aqurecords.at(aquanconRecordIdx).i2; i++)
-                        aqurecords.at(aquanconRecordIdx).global_index_per_record.push_back
-                                                            (
-                                                                grid.getGlobalIndex(i-1, j-1, k-1)
-                                                            );
+                    for (int i=aqurecords.at(aquanconRecordIdx).i1; i <= aqurecords.at(aquanconRecordIdx).i2; i++) {
+                        if (cellActive(grid, i, j, k) && !neighborCellDirectionActive(grid, i, j, k, faceDir) ) {
```

Comment:
If you are making changes anyway I would really prefer if you do the `-1` shift when storing in the `aqurecord` variable; the convention in the rest of the OPM code is that grid indices are converted to offset 0 *immediately*.